packages/runtime/src/state/modules/api-resources.ts

```diff
@@ -1,13 +1,26 @@
 import { APIResource, APIResourceType } from '../../api'
 import deepEqual from '../../utils/deepEqual'
 import { Action, ActionTypes } from '../actions'
+import { Branded } from '../../utils/branded'
 
-type State = Map<APIResourceType, Map<string, APIResource | null>>
+type CompositeResourceId = Branded<string, 'CompositeResourceId'>
+
+function compositeId(resourceId: string, locale?: string | null): CompositeResourceId {
+  return (locale != null ? `${resourceId}@${locale}` : resourceId) as CompositeResourceId
+}
+
+function parseCompositeId(compositeId: CompositeResourceId): [string, string | undefined] {
+  const [resourceId, locale] = compositeId.split('@')
+  return [resourceId, locale]
+}
+
+type State = Map<APIResourceType, Map<CompositeResourceId, APIResource | null>>
```

Comment:
Use "composite" resource ID (resource ID + locale) to make it possible to store "localized" API resources such as `PagePathnameSlice`/`LocalizedGlobalElement` across several different locales.

Reply:
This is a backward-compatible change.

Reply:
My gut reaction says that this should be done with another level of Maps, where the keys are the locales or `null`, but I also assume you had a reason to favor this path. Can you fill me in?

Reply:
@arvinpoddar Good question, the main reason for "flattening" the `Map` hierarchy here is that the extra level of nesting makes both the state updates and the serialization/deserialization code much more verbose/harder to understand. 

The state updates part can probably be redeemed if we pull in [Immer](https://immerjs.github.io/immer/), but serialization/deserialization will still be messier w/ the extra level of the nesting.

Reply:
Yeah, writing state updates with 2+ degrees of nesting is painful. I'm wondering if this causes us to lose the ability to cleanly/efficiently write certain selectors for resources, such as if my selector wants to retrieve from state using only one part of the composite ID. My imagination is failing me with examples of where it would be awful, but just something to think about. I'm not really opposed to this.

Reply:
We can always rewrite this if ID composition becomes an issue in the future, it's just an implementation detail. I do like how little code I had to change here comparing to the alternative.

Reply:
I would love to come up with a Map data structure that does this for us. We will be having some pretty gnarly composite keys for element trees where the key is a combination of `type`, `key`, `locale`, and `siteVersion`. Being able to do something like `map.set([type, key, locale, siteVersion], elementTree)` and transparently get good serialization/deserialization, debug visuals of keys, etc. would be great.

Reply:
Also in this case it would allow us to flatten it and avoid the extra layer with the API resource type

Reply:
Some prior art on this: https://hackernoon.com/composite-keys-a-guide-on-how-to-handle-them